core/metrics/src/main/scala/org/typelevel/otel4s/metrics/ObservableInstrumentBuilder.scala

```diff
@@ -61,4 +61,20 @@ trait ObservableInstrumentBuilder[F[_], A, Instrument] {
   def createWithCallback(
       cb: ObservableMeasurement[F, A] => F[Unit]
   ): Resource[F, Instrument]
+
+  /** Creates an asynchronous instrument based on an effect that produces a
+    * number of measurements.
+    *
+    * The measurement effect will be evaluted when the instrument is being
+    * observed.
+    *
+    * The measurement effect is expected to abide by the following restrictions:
+    *   - Short-living and (ideally) non-blocking
+    *   - Run in a finite amount of time
+    *   - Safe to call repeatedly, across multiple threads
+    *
+    * @param measurements
+    *   Effect that produces a number of measurements
+    */
+  def create(measurements: F[List[Measurement[A]]]): Resource[F, Instrument]
```

Comment:
One late thought: does it need to be a `List` as opposed to anything we can iterate?

Reply:
I made it `List` since I traversed it in the default implementation (that is now removed), but could probably have used something else. These callbacks are called very rarely so performance is probably not a concern.